folium/features.py

```diff
@@ -393,121 +398,167 @@ class GeoJson(Layer):
     """
     _template = Template(u"""
         {% macro script(this, kwargs) %}
+        {%- if this.style %}
+        function {{ this.get_name() }}_styler(feature) {
+            switch({{ this.feature_identifier }}) {
+                {%- for style, ids_list in this.style_map.items() if not style == 'default' %}
+                {% for id_val in ids_list %}case "{{ id_val }}": {% endfor %}
```

Comment:
Hey @Conengmo, starting to work my way through this now. I'll leave comments as I go. This a library-wide design pattern I suppose, but for interpolating literals into the JS templates, how would you feel about using Jinja2's `tojson` method instead of hardcoding in the string quotes around the interpolaters. This would have the added benefit of raising an error (I think) at the python runtime if an object is not JSON serializable.

So this line might look like:
```
{% for id_val in ids_list %}case {{ id_val | tojson | safe }}: {% endfor %}
```
Docs on this feature:

http://flask.pocoo.org/docs/0.12/templating/#standard-filters

Let me know what you think. I think it could be a good option to standardize as a sort of consistent style guide across different classes - and allow us to flexibly, but consistently embed dtypes including strings as well as others (numbers, lists, dicts) with less template hardcoding.

Reply:
Hi Jason, nice idea! Let's do this throughout the project. Maybe make our own filters that do the various mappings? Or is `tojson` suitable for all? If it's okay with you I think we should do that in a separate PR. Maybe you want to open an issue for this?

Reply:
Yes, that sounds great. I'll work in that after we get through getting this merged.